std/experimental/allocator/building_blocks/free_list.d

```diff
@@ -759,23 +766,24 @@ struct ContiguousFreeList(ParentAllocator,
     alias A = ContiguousFreeList!(NullAllocator, 0, 64);
     auto a = A(new ubyte[1024]);
 
-    assert((() nothrow @safe @nogc => a.empty)() == Ternary.yes);
-
-    assert((() pure nothrow @safe @nogc => a.goodAllocSize(15))() == 64);
-    assert((() pure nothrow @safe @nogc => a.goodAllocSize(65))()
-            == (() nothrow @safe @nogc => NullAllocator.instance.goodAllocSize(65))());
-
-    auto b = a.allocate(100);
-    assert((() nothrow @safe @nogc => a.empty)() == Ternary.yes);
-    assert(b.length == 0);
-    // Ensure deallocate inherits from parent
-    () nothrow @nogc { a.deallocate(b); }();
-    b = a.allocate(64);
-    assert((() nothrow @safe @nogc => a.empty)() == Ternary.no);
-    assert(b.length == 64);
-    assert((() nothrow @safe @nogc => a.owns(b))() == Ternary.yes);
-    assert((() nothrow @safe @nogc => a.owns(null))() == Ternary.no);
-    () nothrow @nogc { a.deallocate(b); }();
+    () nothrow @safe @nogc {
+        assert(a.empty == Ternary.yes);
+        assert((() pure => a.goodAllocSize(15))() == 64);
+        assert((() pure => a.goodAllocSize(65))()
+                == NullAllocator.instance.goodAllocSize(65));
+
+        auto b = a.allocate(100);
+        assert(a.empty == Ternary.yes);
+        assert(b.length == 0);
+        // Ensure deallocate inherits from parent
+        () @trusted { a.deallocate(b); }();
```

Comment:
Better close the lambda here and open it again -> you can avoid the ugly `@trusted`